File: src/dataset/python/perfect-numbers/gemini/GeminiFlash_perfect-numbers_v2.py
def classify(number):
    if number <= 0:
        raise ValueError("The value is inferior or equal to zero which is not a natural number")

    aliquot = 0
    for i in range(1, int(number**0.5) + 1):
        if number % i == 0:
            if i * i != number and i != 1:
                aliquot += i + number // i
            elif i == 1:
                aliquot += 1
            elif i * i == number:
                aliquot += i

    if aliquot == number:
        return 'perfect'
    elif aliquot > number:
        return 'abundant'
    else:
        return 'deficient'

File: src/dataset/javascript/perfect-numbers_exercism-javascript-ThomasZumsteg/gemini/GeminiFlash_perfect-numbers_v2.js
function factors(number) {
    if (number <= 1) {
        return [];
    }

    const result = [1];
    for (let f = 2; f * f <= number; f++) {
        if (number % f === 0) {
            result.push(f);
            const otherFactor = number / f;
            if (f !== otherFactor) {
                result.push(otherFactor);
            }
        }
    }
    return result.sort((a, b) => a - b);
}

class PerfectNumbers {
    classify(number) {
        if (number <= 0) {
            return "Classification is only possible for natural numbers.";
        }

        let factorSum = 0;
        const factorsArray = factors(number);
        for (let i = 0; i < factorsArray.length; i++) {
            factorSum += factorsArray[i];
        }

        if (number < factorSum) {
            return "abundant";
        } else if (number > factorSum) {
            return "deficient";
        } else {
            return "perfect";
        }
    }
}

export default PerfectNumbers;